File: tutorials/golang/07-hello-prometheus-metrics/main.go
package main

import (
	"fmt"
	"net/http"
)

func main() {
	mux := newMetricsMux()
	addr := ":4061"
	fmt.Printf("Listening on %s\n", addr)
	fmt.Println("  GET /        — index")
	fmt.Println("  GET /work    — do work (increments counter)")
	fmt.Println("  GET /metrics — Prometheus metrics")
	if err := http.ListenAndServe(addr, mux); err != nil {
		fmt.Printf("Error: %v\n", err)
	}
}

File: tutorials/rust/07-hello-prometheus-metrics/src/main.rs
use hello_prometheus_metrics::app;
use tokio::net::TcpListener;

#[tokio::main]
async fn main() {
    let listener = TcpListener::bind("0.0.0.0:3000").await.unwrap();
    println!("Listening on http://0.0.0.0:3000");
    println!("  POST /work     - do work (increments counter)");
    println!("  GET  /metrics  - prometheus metrics");
    axum::serve(listener, app()).await.unwrap();
}
